### ISSUE TITLE
Error while shutting down failed pipeline.

### ISSUE THREAD

**Issue by ryzhyk** [feldera/feldera#2432]
The `shutdown_pipeline` method doesn't work correctly when the pipeline is in the Failed state. The method sends a shutdown request to the pipeline and starts checking its status. If the status is Failed, it assumes that the operation failed and raises an exception. However, if the pipeline was in the Failed state to start with, it might simply need more time to complete the shutdown operation. (Clarification: the Shutdown command is valid in the Failed state and should change the state of the pipeline from Failed to Shutdown)

In general, the shutdown operation should rarely fail. The one scenario I can think of (other than losing connectivity to Feldera) is a race condition when the pipeline that was in a good state fails for an unrelated after receiving the shutdown command and goes to a failed state, in which case re-sending the command should do the trick. 

In summary, this method should probably just keep polling for, say, 15 seconds if the pipeline is in the failed state, and then maybe re-send the shutdown command again once more and poll a bit longer before giving up. 

https://github.com/feldera/feldera/blob/0bb66f0c3834f4aa121a3938087d5b51bdc711ae/python/feldera/rest/feldera_client.py#L234
